urnx/build.go

```diff
@@ -0,0 +1,34 @@
+package urnx
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+// Build create a new URN with the specified fields
+func Build(namespace string, resourceType string, resourceID uuid.UUID) (*URN, error) {
```

Comment:
Philosophical question: Is this function necessary? `net/url` doesn't provide a constructor for URLs, but it does provide a parse function: https://pkg.go.dev/net/url#URL

I'm fine leaving it if we think it's useful.

Reply:
I think it's useful (but I'm probably biased). I figure having this saves a `newURN` function be built within any number of the infratographer apps